src/corva/models/stream/raw.py

```diff
@@ -96,41 +103,40 @@ def from_raw_event(event: List[dict]) -> List[RawStreamEvent]:
 
         return result
 
-    @staticmethod
+    def get_cached_max_record_value(self, cache: RedisState) -> Optional[float]:
+        result = cache.load(key=self._max_record_value_cache_key)
+
+        if result is None:
+            return result
+
+        return float(result)
+
+    def set_cached_max_record_value(self, cache: RedisState) -> None:
+        cache.store(key=self._max_record_value_cache_key, value=self.max_record_value)
+
     def filter_records(
-        event: RawStreamEvent,
-        last_value: Optional[float],
+        self,
+        old_max_record_value: Optional[float],
     ) -> List[RawBaseRecord]:
-        new_records = copy.deepcopy(event.records)
+        new_records = copy.deepcopy(self.records)
 
-        if event.is_completed:
+        if self.is_completed:
             # there can be only 1 completed record, always located at the end
             new_records = new_records[:-1]  # remove "completed" record
 
-        if last_value is None:
+        if old_max_record_value is None:
             return new_records
 
-        values = [record.main_value for record in new_records]
+        values = [record.record_value for record in new_records]
 
         new_records = [
-            record for record, value in zip(new_records, values) if value > last_value
+            record
+            for record, value in zip(new_records, values)
+            if value > old_max_record_value
         ]
 
         return new_records
 
-    _require_at_least_one_record = pydantic.root_validator(
-        pre=False, skip_on_failure=True, allow_reuse=True
-    )(validators.require_at_least_one_record)
-
-    @pydantic.root_validator(pre=False, skip_on_failure=True)
-    def require_app_key_in_metadata_apps(cls, values):
```

Comment:
Why this validator was removed?

Reply:
It is validated by pydantic automatically now https://github.com/corva-ai/python-sdk/blob/08e004970c861d7e2cfb18af62ab9bf6489d3c7c/src/corva/models/stream/raw.py#L57